service/internal/basedomain/base_domain_test.go

```diff
@@ -0,0 +1,119 @@
+package basedomain
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/giantswarm/operatorkit/controller/context/cachekeycontext"
+
+	"github.com/giantswarm/cluster-operator/service/internal/unittest"
+)
+
+func Test_BaseDomain_Cache(t *testing.T) {
+	testCases := []struct {
+		name             string
+		ctx              context.Context
+		baseDomain       string
+		expectCaching    bool
+		expectBaseDomain string
+	}{
+		{
+			name:             "case 0",
+			ctx:              cachekeycontext.NewContext(context.Background(), "1"),
+			baseDomain:       "domain.company.com",
+			expectCaching:    true,
+			expectBaseDomain: "domain.company.com",
+		},
+		// This is the case where we modify the AWSCluster CR in order to change the
+		// baseDomain value, while the operatorkit caching mechanism is disabled.
+		{
+			name:             "case 1",
+			ctx:              context.Background(),
+			baseDomain:       "olddomain.company.com",
+			expectCaching:    false,
+			expectBaseDomain: "newdomain.company.com",
+		},
+		{
+			name:             "case 2",
+			ctx:              cachekeycontext.NewContext(context.Background(), "1"),
+			baseDomain:       "domain.company.com",
+			expectCaching:    true,
+			expectBaseDomain: "domain.company.com",
+		},
```

Comment:
In case this is the exact same as case 0 we can just drop it. I think there was some different setting for the pod CIDR service where this came from. 

```suggestion
```